### PR TITLE
UpdateQuery: marshal false as BOOLEAN if `nullzero` tag not set

### PR DESCRIPTION
**Pull request by bevzzz**
Related to: #922, see [this comment](https://github.com/uptrace/bun/issues/922#issuecomment-1830374219) for detailed analysis. [This commit](https://github.com/uptrace/bun/commit/56cb98db2aac4db9f78151f157af3e9cf417e598) demonstrates the bug.

Setting `NullZero = true` for fields with `default:x` tag led to `false` values being appended as `NULL` in `UpdateQuery`. 

https://github.com/uptrace/bun/blob/8a4383505d7e954897b23811a412b9cdafaf41eb/schema/table.go#L378-L381
 
Apart from causing the bug, this seemed like a misuse of `NullZero` property. Removing this behaviour required additional (small) changes to `InsertQuery`, so I also added new test cases and extended an existing one to make sure we don't introduce any regressions (hence the many changed files).

Finally, I moved the chunky `f.IsPtr && f.HasNilValue(strct), f.HasZeroValue(strct) && (f.NullZero || f.SQLDefault != "")` conditional to a separate method and added some documentation for it.

### CODE REVIEW

File: internal/dbtest/query_test.go [hunk 1]
```diff
@@ -331,20 +331,27 @@ func TestQuery(t *testing.T) {
...
 		func(db *bun.DB) schema.QueryAppender {
+			// "nullzero" marshals zero values as DEFAULT or NULL (if DEFAULT placeholder is not supported)
+			// DB drivers which support DEFAULT placeholder resolve it to NULL for columns that do not have a DEFAULT value.
 			type Model struct {
 				Int  int64     `bun:",nullzero"`
 				Uint uint64    `bun:",nullzero"`
 				Str  string    `bun:",nullzero"`
 				Time time.Time `bun:",nullzero"`
+				Bool bool      `bun:",nullzero"`
+				// EmptyStr string    `bun:",nullzero"` -- same as Str
```
Comment (vmihailenco) [inline internal/dbtest/query_test.go:342]: Do you want to uncomment this?

**Comment by vmihailenco** [comment]
@bevzzz could you please rebase this?

**Review by vmihailenco — APPROVED** [review]

File: internal/dbtest/query_test.go (hunk 1 above)
Comment (bevzzz) [reply internal/dbtest/query_test.go]: Yes, thank you

**Comment by bevzzz** [comment]
There will be some more conflicts here, because of the test cases added in the other PR we've just merged.

I'll rebase later today 👌

**Comment by vmihailenco** [comment]
@bevzzz thanks :+1: Sorry for the hassle.

**Comment by bevzzz** [comment]
No problem, glad I could contribute :)

**Pull request merged**